drivers/video/arducam_mega.c

```diff
@@ -0,0 +1,1064 @@
+#define CAM_REG_SHARPNESS_CONTROL             0X28
+#define CAM_REG_AUTO_FOCUS_CONTROL            0X29
+#define CAM_REG_IMAGE_QUALITY                 0x2A
+#define CAM_REG_EXPOSURE_GAIN_WHILEBAL_ENABLE 0X30
+#define CAM_REG_MANUAL_GAIN_BIT_9_8           0X31
+#define CAM_REG_MANUAL_GAIN_BIT_7_0           0X32
+#define CAM_REG_MANUAL_EXPOSURE_BIT_19_16     0X33
+#define CAM_REG_MANUAL_EXPOSURE_BIT_15_8      0X34
+#define CAM_REG_MANUAL_EXPOSURE_BIT_7_0       0X35
+#define CAM_REG_BURST_FIFO_READ_OPERATION     0X3C
+#define CAM_REG_SINGLE_FIFO_READ_OPERATION    0X3D
+#define CAM_REG_SENSOR_ID                     0x40
+#define CAM_REG_YEAR_SDK                      0x41
+#define CAM_REG_MONTH_SDK                     0x42
+#define CAM_REG_DAY_SDK                       0x43
+#define CAM_REG_SENSOR_STATE                  0x44
+#define CAM_REG_FPGA_VERSION_NUMBER           0x49
+#define CAM_REG_DEBUG_DEVICE_ADDRESS          0X0A
+#define CAM_REG_DEBUG_REGISTER_HIGH           0X0B
+#define CAM_REG_DEBUG_REGISTER_LOW            0X0C
+#define CAM_REG_DEBUG_REGISTER_VALUE          0X0D
+
+#define SENSOR_STATE_IDLE   (1 << 1)
+#define SENSOR_RESET_ENABLE (1 << 6)
+
+#define CTR_WHILEBALANCE 0X02
+#define CTR_EXPOSURE     0X01
+#define CTR_GAIN         0X00
+
+/**
+ * @struct mega_sdk_data
+ * @brief Basic information of the camera firmware
+ */
+struct mega_sdk_data {
+	uint8_t year;
+	uint8_t month;
+	uint8_t day;
+	uint8_t version;
+};
+
+struct arducam_mega_config {
+	struct spi_dt_spec bus;
+};
+
+struct arducam_mega_data {
+	const struct device *dev;
+	struct video_format fmt;
+	struct k_fifo fifo_in;
+	struct k_fifo fifo_out;
+	struct k_work_delayable buf_work;
+	struct k_work_sync work_sync;
+	struct k_poll_signal *signal;
+	struct arducam_mega_info *info;
+	struct mega_sdk_data ver;
+	uint8_t fifo_first_read;
+	uint32_t fifo_length;
+	uint8_t stream_on;
+};
+
+static struct arducam_mega_info mega_infos[] = {{
+							.support_resoultion = 7894,
+							.support_special_effects = 63,
+							.exposure_value_max = 30000,
+							.exposure_value_min = 1,
+							.gain_value_max = 1023,
+							.gain_value_min = 1,
+							.enable_focus = 1,
+							.enable_sharpness = 0,
+							.device_address = 0x78,
+						},
+						{
+							.support_resoultion = 7638,
+							.support_special_effects = 319,
+							.exposure_value_max = 30000,
+							.exposure_value_min = 1,
+							.gain_value_max = 1023,
+							.gain_value_min = 1,
+							.enable_focus = 0,
+							.enable_sharpness = 1,
+							.device_address = 0x78,
+						}};
+
+#define ARDUCAM_MEGA_VIDEO_FORMAT_CAP(width, height, format)                                       \
+	{                                                                                          \
+		.pixelformat = (format), .width_min = (width), .width_max = (width),               \
+		.height_min = (height), .height_max = (height), .width_step = 0, .height_step = 0  \
+	}
+
+static struct video_format_cap fmts[] = {
```

Comment:
Make const

Reply:
@bjarki-trackunit 
Thank you for your detail check. In fact, the arducam mega driver is used for multi types camera instead of one specific sensor, so the content of the fmts will be changed when connect different cameras which mega driver has supported. That's why we define it in static struct video_format_cap.

Reply:
ok